dataplane/api-server/src/config.rs

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+*/
+use clap::Parser;
+use std::path::PathBuf;
+
+#[derive(Debug, Parser, Clone)]
+pub struct GrpcConfig {
+    #[clap(long, default_value = "false")]
+    pub enable_tls: bool,
+    #[clap(long, default_value = "false")]
+    pub enable_mtls: bool,
+    pub certificate_authority_root_path: Option<PathBuf>,
+    pub server_certificate_path: Option<PathBuf>,
+    pub server_private_key_path: Option<PathBuf>,
+    pub client_certificate_authority_root_path: Option<PathBuf>,
+    pub client_certificate_path: Option<PathBuf>,
+    pub client_private_key_path: Option<PathBuf>,
+}
+
+impl GrpcConfig {
+    pub fn validate(&self) -> Result<(), String> {
```

Comment:
i couldn't find where this is being called. please ignore if clap calls this automatically for us.

Reply:
I swore in my reading, I saw clap would use this method for us. But, alas, I cannot find it in docs. I added an explicit call for this now.

If you know of a better way to get clap to perform validation at parse time, please let me known.